pkg/review/commit_review_status.go

```diff
@@ -192,12 +207,59 @@ func (fn *CommitApprovalDoFn) ProcessElement(ctx context.Context, commit Commit,
 	emit(commitReviewStatus)
 }
 
+// ProcessElement is a DoFn implementation that takes a CommitReviewStatus
+// and populates its breakGlassIssue field (if necessary) and then emits it
+// using the given emit function. The process only searches for break glass
+// issues for commits that do not have the status GithubPRApproved.
+func (fn *BreakGlassIssueDoFn) ProcessElement(ctx context.Context, commitReviewStatus CommitReviewStatus, emit func(status CommitReviewStatus)) {
+	logger := logging.FromContext(ctx)
+	logger.InfoContext(ctx, "processing commitReviewStatus", "commitReviewStatus", commitReviewStatus)
+	if commitReviewStatus.ApprovalStatus != GithubPRApproved {
+		// if the commit does not have proper approval, we check if there was a
+		// break glass issue opened by the author during the timeframe they
+		// submitted the commit.
+		breakGlassIssue, err := getBreakGlassIssue(
+			ctx,
+			fn.Client,
+			fn.Config.IssuesTable,
+			commitReviewStatus.Author,
+			commitReviewStatus.Timestamp,
+		)
+		if err != nil {
+			// We should only get transient style errors from BigQuery
+			// (e.g. network is down etc.). So, we can just log the error and then
+			// drop this CommitReviewStatus from the pipeline. It will then get
+			// retried on the next run of the pipeline.
+			logger.ErrorContext(ctx, "failure when trying to get break glass issue: %v", err)
+			return
+		}
+		commitReviewStatus.BreakGlassURL = breakGlassIssue.HTMLURL
+	}
+	emit(commitReviewStatus)
```

Comment:
We should check if `emit` is not nil before calling.

Reply:
The `emit` function supplied by the Apache Beam runner and will never be nil, so I don't think this is necessary.

Reply:
Defensive programming is always necessary :smile:. Without it, one day their API will change and we'll waste hours chasing down a NPE.

Reply:
A `nil` emit function would be such a wrong thing to pass in that I don't think silently ignoring it and effectively just dropping all commits would be preferable. I added a nil check and some logging in the nil case so that if this does ever happen we have some indication that a major assumption of the pipeline has been violated.

Reply:
In that case, I'd still expect us to catch the error and raise a specific `panic` instead of the default generic panic that Go emits (which can be very difficult to track down).

Reply:
I see, I'm not too familiar with panics. Would that look something like this?
```
if emit == nil {
    panic("nil emit function was provided")
}
```

Reply:
I've updated the function to panic immediately if a `nil` emit function is passed in.